compiler/rustc_trait_selection/src/traits/mod.rs

```diff
@@ -235,7 +237,15 @@ fn do_normalize_predicates<'tcx>(
     // them here too, and we will remove this function when
     // we move over to lazy normalization *anyway*.
     tcx.infer_ctxt().ignoring_regions().enter(|infcx| {
-        let predicates = match fully_normalize(&infcx, cause, elaborated_env, predicates) {
+        let predicates = match predicates.try_map_id(|predicate| {
```

Comment:
Thanks for reviewing! I renamed it to `normalized_predicates`. I think It maybe not easy to use a mutating, since `fully_normalize` needs to take ownership of the elements in predicts. Similar methods that act on instances of `TypeFoldable` have this feature, and `try_map_id` is used to implement `TypeFoldable` for `Vec`.

Reply:
Also, could you make a separate variable for the `try_map_id` call?

Reply:
Okay. I renamed it to `origin_predicate`. We can pick what we want and sqaush them to one commit I think